enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/OnboardingServiceImpl.java

```diff
@@ -247,14 +250,32 @@ public Response performCleanup(OnboardingCleanupRequest request) throws Onboardi
         return new Response();
     }
 
+    /**
+     * Verify process identifier and lock the process until the end of the transaction.
+     * @param ownerId Owner identification.
+     * @param processId Process identifier from request.
+     * @param onboardingStatus Expected onboarding process status.
+     * @throws OnboardingProcessException Thrown in case process identifier is invalid.
+     */
+    public void verifyProcessIdAndLock(OwnerId ownerId, String processId, OnboardingStatus onboardingStatus) throws OnboardingProcessException {
+        final OnboardingProcessEntity process = onboardingProcessRepository.findExistingProcessForActivationWithLock(ownerId.getActivationId(), onboardingStatus)
+                .orElseThrow(() -> new OnboardingProcessException("Onboarding process not found, activation ID: " + ownerId.getActivationId()));
+        String expectedProcessId = process.getId();
+
+        if (!expectedProcessId.equals(processId)) {
+            logger.warn("Invalid process ID received in request: {}, {}", processId, ownerId);
```

Comment:
Let's use exceptions with context and transform logger to:
```suggestion
            throw new OnboardingProcessException(String.format("Invalid process ID received in request: %s, %s", processId, ownerId));
```

Reply:
Ok.